membrane_curvature/lib/mods.py

```diff
@@ -1,89 +1,105 @@
-import itertools as it
 import numpy as np
 
 
-def grid_map(coords, factor):
-    """ Maps (x,y) coordinates to unit cell in grid.
+def derive_surface(n_cells, selection, max_width_x, max_width_y):
+    """
+    Derive surface from AtomGroup positions.
 
     Parameters
     ----------
-    (x, y):  tuple
-        Value of (x, y) coordinates
-    factor:  float
-        Mappign factor to assign grid.
+    n_cells : int.
+        number of cells in the grid of size `max_width`.
+    selection : AtomGroup.
+        AtomGroup of reference selection to define the surface
+        of the membrane.
+    max_width : float.
+        Maximum width of simulation box. (Determined by simulation box dimensions)
 
-        Returns
-        -------
-        Returns l, m  with l,m as int.
+    Returns
+    -------
+    Returns set of z coordinates 
 
     """
+    coordinates = selection.positions
+    return get_z_surface(coordinates, n_x_bins=n_cells, n_y_bins=n_cells,
+                         x_range=(0, max_width_x), y_range=(0, max_width_y))
 
-    index_grid_l = int(abs(coords[0]) * factor)
-    index_grid_m = int(abs(coords[1]) * factor)
 
-    return index_grid_l, index_grid_m
-
-
-def core_fast_leaflet(universe, z_Ref, n_cells, selection, max_width):
+def get_z_surface(coordinates, n_x_bins=10, n_y_bins=10, x_range=(0, 100), y_range=(0, 100)):
     """
-    Runs core_fast_leaflet for selected surface
+    Derive surface from distribution of z coordinates in grid.
 
     Parameters
     ----------
-    universe: MDA Universe
-        Provides coordinates and trajectory of the coordinates.
-    z_Ref : dict.
-        Dictionary to store values in every iteration over frames.
-    n_cells : int.
-        number of cells in the grid of size `max_width`.
-    selection : AtomGroup
-        AtomGroup of reference selection to define the surface
-        of the membrane.
-    max_width : int.
-        Maximum width of simulation box.
+    coordinates : tuple.
+        Numpy.ndarray with shape=(n_atoms, 3).
+    n_x_bins : int.
+        Number of bins in grid in the x dimension. 
+    n_y_bins : int.
+        Number of bins in grid in the y dimension. 
+    x_range : tuple
+        Range of indexes in grid in the x dimension with shape=(0, max_width_x).
```

Comment:
I still think this would be the minimum x coordinate and the maximum x coordinate in angstrom, instead of the indexes in the grid. It's a replacement for `max_width` rather than n_cells, right?

Reply:
Sorry, I thought I included this change already. Give me a sec to update.